File: backend/internal/domain/entities/listing.go
package entities

import "time"

type ListingType string

const (
	ListingTypeSale ListingType = "sale"
	ListingTypeRent ListingType = "rent"
)

type Listing struct {
	ID        string      `json:"id"`
	Type      ListingType `json:"type"`
	ValueBRL  float64     `json:"value_brl"`
	ImageURL  *string     `json:"image_url,omitempty"`
	Address   Address     `json:"address"`
	CreatedAt time.Time   `json:"created_at"`
}

type ListingFilter struct {
	Type *ListingType
}

File: frontend/lib/modules/listing/domain/entities/listing.dart
import 'address.dart';

enum ListingType { sale, rent }

class Listing {
  const Listing({
    required this.id,
    required this.type,
    required this.valueBRL,
    this.valueUSD,
    this.imageUrl,
    required this.address,
    required this.createdAt,
  });

  final String id;
  final ListingType type;
  final double valueBRL;
  final double? valueUSD;
  final String? imageUrl;
  final Address address;
  final DateTime createdAt;

  String get typeLabel => type == ListingType.sale ? 'sale' : 'rent';
}
